agent/api/task.go

```diff
@@ -500,6 +504,26 @@ func (task *Task) DockerHostConfig(container *Container, dockerContainerMap map[
 	return task.dockerHostConfig(container, dockerContainerMap)
 }
 
+func (task *Task) ApplyExecutionRoleLogsAuth(hostConfig *docker.HostConfig, credentialsManager credentials.Manager) *HostConfigError {
+	id := task.GetExecutionCredentialsID()
+	if id == "" {
+		// No execution credentials set for the task. Do not inject the endpoint environment variable.
+		return &HostConfigError{"No execution credentials set for the task"}
+	}
+
+	executionRoleCredentials, ok := credentialsManager.GetTaskCredentials(id)
+	if !ok {
+		// Task has credentials id set, but credentials manager is unaware of
+		// the id. This should never happen as the payload handler sets
```

Comment:
"should never happen" in a comment is asking for it to happen. 😄 

It looks like this gets called during CreateContainer.  We need to verify (and probably write a test to enforce) that credentials are present before CreateContainer gets invoked.  @richardpen did work here in https://github.com/aws/amazon-ecs-agent/pull/906 for pull, but we need to make sure that the same dependency condition is applied for create as well.

Reply:
I agree on the sentiment here. However, I'd also advise against changing this logic significantly at this point of time. 

We follow a similar logic in `api.Task`'s `PostUnmarshalTask` method. I think returning a `HostConfigError` seems like an acceptable way to handle that edge case.

Reply:
leaving this logic as is for now.